src/test/common/clara/test_node_sharing.cljc

```diff
@@ -0,0 +1,303 @@
+               WindSpeed
+               ColdAndWindy
+               WindSpeed]))
+
+   :cljs
+   (ns clara.test-node-sharing
+     (:require [clara.rules :refer [fire-rules
+                                    insert
+                                    insert!
+                                    insert-all
+                                    insert-unconditional!
+                                    retract
+                                    query]]
+               [clara.rules.testfacts :refer [->Temperature Temperature
+                                              ->Cold Cold
+                                              ->WindSpeed WindSpeed
+                                              ->ColdAndWindy ColdAndWindy]]
+               [clara.rules.accumulators]
+               [cljs.test]
+               [schema.test :as st])
+     (:require-macros [clara.tools.testing-utils :refer [def-rules-test]]
+                      [cljs.test :refer [is deftest run-tests testing use-fixtures]])))
+
+(use-fixtures :once st/validate-schemas #?(:clj tu/opts-fixture) tu/side-effect-holder-fixture)
+
+;; See issue 433 for more information
+(def-rules-test test-or-sharing-same-condition
+  {:rules [or-rule [[[:or
+                      [::a]
+                      [::b]]
+                     [::d]]
+                    (insert! {:fact-type ::c})]
+
+           other-rule [[[::a]
+                        [::d]]
+                       (insert! {:fact-type ::e})]]
+   :queries [c-query [[] [[?c <- ::c]]]
+             e-query [[] [[?e <- ::e]]]]
+
+   :sessions [empty-session [or-rule other-rule c-query e-query] {:fact-type-fn :fact-type}]}
+  (let [session (-> empty-session
+                    (insert {:fact-type ::b})
+                    (insert {:fact-type ::d})
+                    fire-rules)]
+
+    (is (= (set (query session c-query))
+           #{{:?c {:fact-type ::c}}}))
+
+    (is (= (set (query session e-query))
+           #{}))))
+
+;; Replicate the same logical scenario as above in test-or-sharing-same-condition
+;; in a single rule with a boolean :or condition. 
+(def-rules-test test-or-sharing-same-condition-in-same-production
+  {:rules [single-rule [[[:or
+                          [:and [:or
+                                 [::a]
+                                 [::b]]
+                           [::d]]
+                          [:and
+                           [::a]
+                           [::d]]]]
+
+                        (insert! {:fact-type ::c})]]
+
+   :queries [c-query [[] [[?c <- ::c]]]]
+
+   :sessions [empty-session [single-rule c-query] {:fact-type-fn :fact-type}]}
+
+  (let [session (-> empty-session
+                    (insert {:fact-type ::b})
+                    (insert {:fact-type ::d})
+                    fire-rules)]
+
+    (is (= (query session c-query)
+           [{:?c {:fact-type ::c}}]))))
+
+
+;; FIXME: Log an issue for this bug and uncomment when it is resolved.  Since an :or
```

Comment:
This fails both before and after https://github.com/cerner/clara-rules/pull/434 so I don't think it is a regression.